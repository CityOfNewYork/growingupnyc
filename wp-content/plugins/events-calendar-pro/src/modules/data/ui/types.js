/* eslint-disable max-len */
/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_PRO_STORE } from '@moderntribe/events-pro/data/prefix';

export const TOGGLE_REPEAT_EVENTS_BLOCK_VISIBILITY = `${ PREFIX_EVENTS_PRO_STORE }/TOGGLE_REPEAT_EVENTS_BLOCK_VISIBILITY`;

export const TOGGLE_RULE_PANEL_VISIBILITY = `${ PREFIX_EVENTS_PRO_STORE }/TOGGLE_RULE_PANEL_VISIBILITY`;
export const HIDE_RULE_PANEL = `${ PREFIX_EVENTS_PRO_STORE }/HIDE_RULE_PANEL`;

export const TOGGLE_RULE_PANEL_EXPAND = `${ PREFIX_EVENTS_PRO_STORE }/TOGGLE_RULE_PANEL_EXPAND`;
export const EXPAND_RULE_PANEL = `${ PREFIX_EVENTS_PRO_STORE }/EXPAND_RULE_PANEL`;

export const TOGGLE_EXCEPTION_PANEL_VISIBILITY = `${ PREFIX_EVENTS_PRO_STORE }/TOGGLE_EXCEPTION_PANEL_VISIBILITY`;
export const HIDE_EXCEPTION_PANEL = `${ PREFIX_EVENTS_PRO_STORE }/HIDE_EXCEPTION_PANEL`;

export const TOGGLE_EXCEPTION_PANEL_EXPAND = `${ PREFIX_EVENTS_PRO_STORE }/TOGGLE_EXCEPTION_PANEL_EXPAND`;
export const EXPAND_EXCEPTION_PANEL = `${ PREFIX_EVENTS_PRO_STORE }/EXPAND_EXCEPTION_PANEL`;
