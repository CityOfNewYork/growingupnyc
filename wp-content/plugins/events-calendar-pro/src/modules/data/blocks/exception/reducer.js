/**
 * External dependencies
 */

/**
 * Internal dependencies
 */
import * as types from './types';

function edit( state, action ) {
	const field = Object.assign( {}, state[ action.index ], action.payload );

	if ( state.length === 1 ) {
		return [ field ];
	}
	return [
		...state.slice( 0, action.index ),
		field,
		...state.slice( action.index + 1 ),
	];
}

export default function exception( state = [], action ) {
	switch ( action.type ) {
		case types.ADD_EXCEPTION:
			return [
				...state,
				action.payload,
			];
		case types.EDIT_EXCEPTION:
			return edit( state, action );
		case types.REMOVE_EXCEPTION:
			return state.filter( ( _, index ) => index !== action.index );
		case types.SYNC_EXCEPTIONS_FROM_DB:
			return JSON.parse( action.payload );
		default:
			return state;
	}
}
