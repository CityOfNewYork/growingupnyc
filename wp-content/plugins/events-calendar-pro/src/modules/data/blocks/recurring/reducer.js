/**
 * External dependencies
 */

/**
 * Internal dependencies
 */
import * as types from './types';

function edit( state, action ) {
	const field = Object.assign( {}, state[ action.index ], action.payload );

	if ( state.length === 1 ) {
		return [ field ];
	}
	return [
		...state.slice( 0, action.index ),
		field,
		...state.slice( action.index + 1 ),
	];
}

export default function recurring( state = [], action ) {
	switch ( action.type ) {
		case types.ADD_RULE:
			return [
				...state,
				action.payload,
			];
		case types.EDIT_RULE:
			return edit( state, action );
		case types.REMOVE_RULE:
			return state.filter( ( _, index ) => index !== action.index );
		case types.SYNC_RULES_FROM_DB:
			return JSON.parse( action.payload );
		default:
			return state;
	}
}
