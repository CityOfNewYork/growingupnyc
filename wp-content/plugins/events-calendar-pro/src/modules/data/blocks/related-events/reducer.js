/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import * as types from './types';

export const DEFAULT_STATE = {
	title: __( 'Related Events', 'tribe-events-calendar-pro' ),
	displayImages: true,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_RELATED_EVENTS_TITLE:
			return {
				...state,
				title: action.payload.title,
			};
		case types.SET_RELATED_EVENTS_DISPLAY_IMAGES:
			return {
				...state,
				displayImages: action.payload.displayImages,
			};
		default:
			return state;
	}
};
