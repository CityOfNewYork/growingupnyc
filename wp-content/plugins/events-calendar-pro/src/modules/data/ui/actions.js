/**
 * Internal dependencies
 */
import * as types from './types';

export const toggleRepeatBlocksVisibility = () => ( {
	type: types.TOGGLE_REPEAT_EVENTS_BLOCK_VISIBILITY,
} );

export const toggleRulePanelVisibility = () => ( {
	type: types.TOGGLE_RULE_PANEL_VISIBILITY,
} );

export const hideRulePanel = () => ( {
	type: types.HIDE_RULE_PANEL,
} );

export const toggleExceptionPanelVisibility = () => ( {
	type: types.TOGGLE_EXCEPTION_PANEL_VISIBILITY,
} );

export const toggleRulePanelExpand = () => ( {
	type: types.TOGGLE_RULE_PANEL_EXPAND,
} );

export const expandRulePanel = () => ( {
	type: types.EXPAND_RULE_PANEL,
} );

export const hideExceptionPanel = () => ( {
	type: types.HIDE_EXCEPTION_PANEL,
} );

export const expandExceptionPanel = () => ( {
	type: types.EXPAND_EXCEPTION_PANEL,
} );

export const toggleExceptionPanelExpand = () => ( {
	type: types.TOGGLE_EXCEPTION_PANEL_EXPAND,
} );

