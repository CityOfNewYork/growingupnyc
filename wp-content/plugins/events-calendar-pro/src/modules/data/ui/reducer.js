/**
 * Internal dependencies
 */
import * as types from './types';

export const DEFAULT_STATE = {
	isRepeatBlockVisible: false,
	isRulePanelVisible: false,
	isExceptionPanelVisible: false,
	isRulePanelExpanded: false,
	isExceptionPanelExpanded: false,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.TOGGLE_REPEAT_EVENTS_BLOCK_VISIBILITY:
			return {
				...state,
				isRepeatBlockVisible: ! state.isRepeatBlockVisible,
			};
		case types.TOGGLE_RULE_PANEL_VISIBILITY:
			return {
				...state,
				isRulePanelVisible: ! state.isRulePanelVisible,
			};
		case types.HIDE_RULE_PANEL:
			return {
				...state,
				isRulePanelVisible: false,
			};
		case types.TOGGLE_EXCEPTION_PANEL_VISIBILITY:
			return {
				...state,
				isExceptionPanelVisible: ! state.isExceptionPanelVisible,
			};
		case types.HIDE_EXCEPTION_PANEL:
			return {
				...state,
				isExceptionPanelVisible: false,
			};
		case types.TOGGLE_RULE_PANEL_EXPAND:
			return {
				...state,
				isRulePanelExpanded: ! state.isRulePanelExpanded,
			};
		case types.EXPAND_RULE_PANEL:
			return {
				...state,
				isRulePanelExpanded: true,
			};
		case types.TOGGLE_EXCEPTION_PANEL_EXPAND:
			return {
				...state,
				isExceptionPanelExpanded: ! state.isExceptionPanelExpanded,
			};
		case types.EXPAND_EXCEPTION_PANEL:
			return {
				...state,
				isExceptionPanelExpanded: true,
			};
		default:
			return state;
	}
};
