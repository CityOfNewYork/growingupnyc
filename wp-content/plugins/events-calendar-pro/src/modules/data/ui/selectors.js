/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { constants } from '@moderntribe/common/data/plugins';

export const getUI = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ].ui;

export const isRepeatBlockVisible = createSelector(
	[ getUI ],
	( ui ) => ui.isRepeatBlockVisible
);
export const isRulePanelVisible = createSelector(
	[ getUI ],
	( ui ) => ui.isRulePanelVisible
);
export const isExceptionPanelVisible = createSelector(
	[ getUI ],
	( ui ) => ui.isExceptionPanelVisible
);
export const isRulePanelExpanded = createSelector(
	[ getUI ],
	( ui ) => ui.isRulePanelExpanded
);
export const isExceptionPanelExpanded = createSelector(
	[ getUI ],
	( ui ) => ui.isExceptionPanelExpanded
);
