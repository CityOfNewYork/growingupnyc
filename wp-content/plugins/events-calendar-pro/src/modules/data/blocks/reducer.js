/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import recurring from '@moderntribe/events-pro/data/blocks/recurring/reducer';
import exception from '@moderntribe/events-pro/data/blocks/exception/reducer';
import additionalFields from '@moderntribe/events-pro/data/blocks/additional-fields';
import relatedEvents from '@moderntribe/events-pro/data/blocks/related-events';

export default combineReducers( {
	recurring,
	exception,
	additionalFields,
	relatedEvents,
} );
