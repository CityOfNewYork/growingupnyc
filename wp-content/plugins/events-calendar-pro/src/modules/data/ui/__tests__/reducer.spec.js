/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/events-pro/data/ui';
import reducer, { DEFAULT_STATE } from '@moderntribe/events-pro/data/ui/reducer';

describe( '[STORE] - UI reducer', () => {
	it( 'Should return the default state', () => {
		expect( reducer( undefined, {} ) ).toEqual( DEFAULT_STATE );
	} );

	it( 'Should toggle repeat panel', () => {
		const current = reducer( DEFAULT_STATE, actions.toggleRepeatBlocksVisibility() );
		expect( current ).toMatchSnapshot();
	} );

	it( 'Should toggle rule panel', () => {
		const current = reducer( DEFAULT_STATE, actions.toggleRulePanelVisibility() );
		expect( current ).toMatchSnapshot();
	} );

	it( 'Should hide rule panel', () => {
		const current = reducer( { ...DEFAULT_STATE, isRulePanelVisible: true }, actions.hideRulePanel() );
		expect( current ).toMatchSnapshot();
	} );

	it( 'Should toggle exception panel', () => {
		const current = reducer( DEFAULT_STATE, actions.toggleExceptionPanelVisibility() );
		expect( current ).toMatchSnapshot();
	} );

	it( 'Should hide exception panel', () => {
		const current = reducer( { ...DEFAULT_STATE, isExceptionPanelVisible: true }, actions.hideExceptionPanel() );
		expect( current ).toMatchSnapshot();
	} );

	it( 'Should toggle rule panel expand', () => {
		const current = reducer( DEFAULT_STATE, actions.toggleRulePanelExpand() );
		expect( current ).toMatchSnapshot();
	} );

	it( 'Should expand rule panel', () => {
		const current = reducer( DEFAULT_STATE, actions.expandRulePanel() );
		expect( current ).toMatchSnapshot();
	} );

	it( 'Should toggle exception panel expand', () => {
		const current = reducer( DEFAULT_STATE, actions.toggleExceptionPanelExpand() );
		expect( current ).toMatchSnapshot();
	} );

	it( 'Should expand exception panel', () => {
		const current = reducer( DEFAULT_STATE, actions.expandExceptionPanel() );
		expect( current ).toMatchSnapshot();
	} );
} );
