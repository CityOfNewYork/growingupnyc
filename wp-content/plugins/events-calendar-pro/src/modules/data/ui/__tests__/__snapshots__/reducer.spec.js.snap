// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - UI reducer Should expand exception panel 1`] = `
Object {
  "isExceptionPanelExpanded": true,
  "isExceptionPanelVisible": false,
  "isRepeatBlockVisible": false,
  "isRulePanelExpanded": false,
  "isRulePanelVisible": false,
}
`;

exports[`[STORE] - UI reducer Should expand rule panel 1`] = `
Object {
  "isExceptionPanelExpanded": false,
  "isExceptionPanelVisible": false,
  "isRepeatBlockVisible": false,
  "isRulePanelExpanded": true,
  "isRulePanelVisible": false,
}
`;

exports[`[STORE] - UI reducer Should hide exception panel 1`] = `
Object {
  "isExceptionPanelExpanded": false,
  "isExceptionPanelVisible": false,
  "isRepeatBlockVisible": false,
  "isRulePanelExpanded": false,
  "isRulePanelVisible": false,
}
`;

exports[`[STORE] - UI reducer Should hide rule panel 1`] = `
Object {
  "isExceptionPanelExpanded": false,
  "isExceptionPanelVisible": false,
  "isRepeatBlockVisible": false,
  "isRulePanelExpanded": false,
  "isRulePanelVisible": false,
}
`;

exports[`[STORE] - UI reducer Should toggle exception panel 1`] = `
Object {
  "isExceptionPanelExpanded": false,
  "isExceptionPanelVisible": true,
  "isRepeatBlockVisible": false,
  "isRulePanelExpanded": false,
  "isRulePanelVisible": false,
}
`;

exports[`[STORE] - UI reducer Should toggle exception panel expand 1`] = `
Object {
  "isExceptionPanelExpanded": true,
  "isExceptionPanelVisible": false,
  "isRepeatBlockVisible": false,
  "isRulePanelExpanded": false,
  "isRulePanelVisible": false,
}
`;

exports[`[STORE] - UI reducer Should toggle repeat panel 1`] = `
Object {
  "isExceptionPanelExpanded": false,
  "isExceptionPanelVisible": false,
  "isRepeatBlockVisible": true,
  "isRulePanelExpanded": false,
  "isRulePanelVisible": false,
}
`;

exports[`[STORE] - UI reducer Should toggle rule panel 1`] = `
Object {
  "isExceptionPanelExpanded": false,
  "isExceptionPanelVisible": false,
  "isRepeatBlockVisible": false,
  "isRulePanelExpanded": false,
  "isRulePanelVisible": true,
}
`;

exports[`[STORE] - UI reducer Should toggle rule panel expand 1`] = `
Object {
  "isExceptionPanelExpanded": false,
  "isExceptionPanelVisible": false,
  "isRepeatBlockVisible": false,
  "isRulePanelExpanded": true,
  "isRulePanelVisible": false,
}
`;
